modules/data.atmosphere/R/download.GFDL.R

```diff
@@ -0,0 +1,208 @@
+                   CF.name = c("air_temperature","surface_downwelling_longwave_flux_in_air","air_pressure","surface_downwelling_shortwave_flux_in_air","eastward_wind","northward_wind","specific_humidity","precipitation_flux"),
+                   units = c('Kelvin',"W/m2","Pascal","W/m2","m/s","m/s","g/g","kg/m2/s")
+  )
+  #2920 values per year for 3 hourly 
+  
+  if (start >= 20060101 & start< 20101231){
+    start_url = paste0(20060101)
+    end_url = paste0(20101231)
+  }
+  if (start >= 20110101 & start< 20151231){
+    start_url = paste0(20110101)
+    end_url = paste0(20151231)
+  }
+  if (start >= 20160101 & start< 20201231){
+    start_url = paste0(20160101)
+    end_url = paste0(20201231)
+  }
+  if (start >= 20210101 & start< 20251231){
+    start_url = paste0(20210101)
+    end_url = paste0(20251231)
+  }
+  if (start >= 20260101 & start< 20301231){
+    start_url = paste0(20260101)
+    end_url = paste0(20301231)
+  }
+  if (start >= 20310101 & start< 20351231){
+    start_url = paste0(20310101)
+    end_url = paste0(20351231)
+  }
+  if (start >= 20360101 & start< 20401231){
+    start_url = paste0(20360101)
+    end_url = paste0(20401231)
+  }
+  if (start >= 20410101 & start< 20451231){
+    start_url = paste0(20410101)
+    end_url = paste0(20451231)
+  }
+  if (start >= 20460101 & start< 20501231){
+    start_url = paste0(20460101)
+    end_url = paste0(20501231)
+  }
+  if (start >= 20510101 & start< 20551231){
+    start_url = paste0(20510101)
+    end_url = paste0(20551231)
+  }
+  if (start >= 20560101 & start< 20601231){
+    start_url = paste0(20560101)
+    end_url = paste0(20601231)
+  }
+  if (start >= 20610101 & start< 20651231){
+    start_url = paste0(20610101)
+    end_url = paste0(20651231)
+  }
+  if (start >= 20660101 & start< 20701231){
+    start_url = paste0(20660101)
+    end_url = paste0(20701231)
+  }
+  if (start >= 20710101 & start< 20751231){
+    start_url = paste0(20710101)
+    end_url = paste0(20751231)
+  }
+  if (start >= 20760101 & start< 20801231){
+    start_url = paste0(20760101)
+    end_url = paste0(20801231)
+  }
+  if (start >= 20810101 & start< 20851231){
+    start_url = paste0(20810101)
+    end_url = paste0(20851231)
+  }
+  if (start >= 20860101 & start< 20901231){
+    start_url = paste0(20860101)
+    end_url = paste0(20901231)
+  }
+  if (start >= 20910101 & start< 20951231){
+    start_url = paste0(20910101)
+    end_url = paste0(20951231)
+  }
+  if (start >= 20960101 & start< 21001231){
+    start_url = paste0(20960101)
+    end_url = paste0(21001231)
+  }
+
+  if (start_year%%5 == 1){
+    time_range <- c(1:2920)
+  }
+  if (start_year%%5 == 2){
+    time_range = c(2921:5840)
+  }
+  if (start_year%%5 == 3){
+    time_range = c(5841:8760)
+  }
+  if (start_year%%5 == 4){
+    time_range = c(8761:11680)
+  }
+  if (start_year%%5 == 0){
+    time_range = c(11681:14600)
+  }
+
+  for (i in 1:rows){
+    year = ylist[i]    
```

Comment:
I'm not questioning (for now at least) the inefficiency of downloading the same file 5 times to download 5 years. I'm questioning that you have a for loop over years, that just downloads the same year (start_year) every single time through the loop. Worse, the only place the variable 'year' is used is in the results return, so the database is incorrectly let to believe that the years are different even though they are all the same.  So for example if I set start_year to 2006 and end year to 2100, I'd just get 95 files that all contain the 2006 data but are labeled 2006 through 2100.  The start_url, end_url, and time_range determine which years you download, so that needs to be INSIDE you loop over years.

FWIW, you could avoid downloading the same file 5 times by looping over files first, then years within files, but I'm OK with that being a later pull request